html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm

```diff
@@ -145,7 +154,15 @@ sub oauth2Result : Path : Args(1) {
     my $token;
 
     eval {
-        $token = $self->oauth2_client($c,$provider)->get_access_token($code);
+        if($provider eq "twitter"){
```

Comment:
Put a space before and after a condtion and use single quotes for constant strings.